pkg/asset/agent/common.go

```diff
@@ -1,18 +1,50 @@
 package agent
 
 import (
+	"github.com/openshift/installer/pkg/types"
 	"github.com/openshift/installer/pkg/types/baremetal"
 	"github.com/openshift/installer/pkg/types/none"
 	"github.com/openshift/installer/pkg/types/vsphere"
+
+	hiveext "github.com/openshift/assisted-service/api/hiveextension/v1beta1"
 )
 
-// SupportedPlatforms lists the supported platforms for agent installer
-var SupportedPlatforms = []string{baremetal.Name, vsphere.Name, none.Name}
+// SupportedInstallerPlatforms lists the supported platforms for agent installer
+func SupportedInstallerPlatforms() []string {
+	return []string{baremetal.Name, vsphere.Name, none.Name}
+}
+
+var supportedHivePlatforms = []hiveext.PlatformType{
```

Comment:
I see that these are also defined for infrastructure types here: https://github.com/openshift/installer/blob/master/vendor/github.com/openshift/api/config/v1/types_infrastructure.go#L177
Would it make sense to use the infrastructure defines instead of the the Hive defines (same strings)?

Reply:
No, because they're different types and casting is how we got into this :smile: